lib/timeline/component/ItemSet.js

```diff
@@ -2070,69 +2084,65 @@ ItemSet.prototype._onAddItem = function (event) {
+      end = this.body.util.toTime(x + this.props.width / 5);
+      newItemData.end = snap ? snap(end, scale, step) : end;
     }
-    // var xAbs = util.getAbsoluteLeft(this.dom.frame);
-    // var x = event.center.x - xAbs;
-    var start = this.body.util.toTime(x);
-    var scale = this.body.util.getScale();
-    var step = this.body.util.getStep();
-    var end;
-
-    var newItemData;
-    if (event.type == 'drop') {
-      newItemData = JSON.parse(event.dataTransfer.getData("text"));
-      newItemData.content = newItemData.content ? newItemData.content : 'new item';
-      newItemData.start = newItemData.start ? newItemData.start : (snap ? snap(start, scale, step) : start);
-      newItemData.type = newItemData.type || 'box';
-      newItemData[this.itemsData._fieldId] = newItemData.id || util.randomUUID();
-
-      if (newItemData.type == 'range' && !newItemData.end) {
-        end = this.body.util.toTime(x + this.props.width / 5);
-        newItemData.end = snap ? snap(end, scale, step) : end;
-      }
-    } else {
-      newItemData = {
-        start: snap ? snap(start, scale, step) : start,
-        content: 'new item'
-      };
-      newItemData[this.itemsData._fieldId] = util.randomUUID();
-
-      // when default type is a range, add a default end date to the new item
-      if (this.options.type === 'range') {
-        end = this.body.util.toTime(x + this.props.width / 5);
-        newItemData.end = snap ? snap(end, scale, step) : end;
-      }
+  } else {
+    newItemData = {
+      start: snap ? snap(start, scale, step) : start,
+      content: 'new item'
+    };
+    newItemData[this.itemsData._fieldId] = util.randomUUID();
+
+    // when default type is a range, add a default end date to the new item
+    if (this.options.type === 'range') {
+      end = this.body.util.toTime(x + this.props.width / 5);
+      newItemData.end = snap ? snap(end, scale, step) : end;
     }
+  }
 
-    var group = this.groupFromTarget(event);
-    if (group) {
-      newItemData.group = group.groupId;
-    }
+  var group = this.groupFromTarget(event);
+  if (group) {
+    newItemData.group = group.groupId;
+  }
 
-    // execute async handler to customize (or cancel) adding an item
-    newItemData = this._cloneItemData(newItemData);     // convert start and end to the correct type
-    this.options.onAdd(newItemData, function (item) {
-      if (item) {
-        me.itemsData.getDataSet().add(item);
-        if (event.type == 'drop') {
-          me.setSelection([item.id]);
-        }
-        // TODO: need to trigger a redraw?
+  // execute async handler to customize (or cancel) adding an item
+  newItemData = this._cloneItemData(newItemData);     // convert start and end to the correct type
+  this.options.onAdd(newItemData, function (item) {
+    if (item) {
+      me.itemsData.getDataSet().add(item);
+      if (event.type == 'drop') {
+        me.setSelection([item.id]);
       }
-    });
-  }
+      // TODO: need to trigger a redraw?
```

Comment:
Was this TODO to be addressed prior to merge? Or would this be expected to be determined after merge?

Reply:
I was there before this PR. Still not sure if a redraw trigger is needed. 
I'm trying to fully understand the redrawing flows to understand how to improve the flow